src/main/java/org/jenkinsci/plugins/workflow/graphanalysis/ForkScanner.java

```diff
@@ -153,11 +151,13 @@ public boolean apply(@Nullable FlowNode input) {
 
     /** Originally a workaround to deal with needing the {@link StepDescriptor} to determine if a node is a parallel start
      *  Now tidily solved by {@link IsParallelStartPredicate}*/
-    private static Predicate<FlowNode> parallelStartPredicate = new IsParallelStartPredicate();
+    private static final Predicate<FlowNode> parallelStartPredicate = new IsParallelStartPredicate();
 
-    // Invoke this passing a test against the ParallelStep conditions
+    /** Now a complete no-op -- originally this was a workaround for dependency issues with workflow-cps.
+     *  Specifically, requiring classes from workflow-cps to detect if something is a parallel step.
+     */
+    @Deprecated
```

Comment:
As a matter of procedure I would suggest doing that ahead of time with a snapshot dependency to make sure you are not forgetting something in this PR.